### PR TITLE
Match CY/FireSim deps | Unpin deps | Update lockfiles

### PR DESCRIPTION
**Pull request by abejgonzalez**
This sync's FireSim's conda environment with Chipyard and at the same time unpins Chipyard dependencies. This should help the solver resolve more often and faster.

**Related PRs / Issues**:

**Type of change**:
- [ ] Bug fix
- [ ] New feature
- [ ] Other enhancement

**Impact**:
- [ ] RTL change
- [ ] Software change (RISC-V software)
- [ ] Build system change
- [ ] Other

**Contributor Checklist**:
- [ ] Did you set `main` as the base branch?
- [ ] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [ ] Did you state the type-of-change/impact?
- [ ] Did you delete any extraneous prints/debugging code?
- [ ] Did you mark the PR with a `changelog:` label?
- [ ] (If applicable) Did you update the conda `.conda-lock.yml` file if you updated the conda requirements file?
- [ ] (If applicable) Did you add documentation for the feature?
- [ ] (If applicable) Did you add a test demonstrating the PR?

- [ ] (If applicable) Did you mark the PR as `Please Backport`?

### CODE REVIEW

File: conda-reqs/chipyard.yaml
Comment (tianrui-wei) [inline conda-reqs/chipyard.yaml:49]: Should we perhaps merge https://github.com/ucb-bar/chipyard/pull/1376 prior to this or combine them in a single PR?

**Review by tianrui-wei — CHANGES_REQUESTED** [review]

File: conda-reqs/chipyard.yaml
Comment (abejgonzalez) [reply conda-reqs/chipyard.yaml:49]: `e2fsprogs` will be removed when the Linux bump happens so lets hold off on that for now.

**Comment by abejgonzalez** [comment]
I got offline approval from @jerryz123. Merging now.

**Pull request merged**
